core/toolbox/toolbox.js

```diff
@@ -542,6 +542,90 @@ Blockly.Toolbox.prototype.setVisible = function(isVisible) {
   this.HtmlDiv.style.display = isVisible ? 'block' : 'none';
 };
 
+/**
+ * Sets the given item as selected.
+ * @param {?Blockly.ToolboxItem} newItem The toolbox item to select.
+ * @public
+ */
+Blockly.Toolbox.prototype.setSelectedItem = function(newItem) {
+  var oldItem = this.selectedItem_;
+
+  if ((!newItem && !oldItem) || (newItem && !newItem.isSelectable())) {
+    return;
+  }
+  newItem = /** @type {Blockly.SelectableToolboxItem} */ (newItem);
+  // Do not deselect if the oldItem has children and has been previously clicked
+  // on.
+  if (oldItem && (!oldItem.isCollapsible() || oldItem != newItem)) {
+    this.selectedItem_ = null;
+    oldItem.setSelected(false);
+    Blockly.utils.aria.setState(/** @type {!Element} */ (this.contentsDiv_),
+        Blockly.utils.aria.State.ACTIVEDESCENDANT, '');
+  }
+
+  if (newItem && newItem != oldItem ) {
```

Comment:
Not that I have been able to figure out. 
Mostly because of this line `(!oldItem.isCollapsible() || oldItem != newItem)`.
This covers the case where we don't deselect a category if it is collapsible.

Reply:
Discussed offline and approved as-is or with a simple comment change.